test/integration/gateway_test.go

```diff
@@ -402,3 +407,275 @@ func TestUnmanagedGatewayClass(t *testing.T) {
 		return false
 	}, gatewayUpdateWaitTime, time.Second)
 }
+
+func TestGatewayFilters(t *testing.T) {
+	ns, cleanup := namespace(t)
+	defer cleanup()
+	other, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name()+"other")
+	require.NoError(t, err)
+	defer func(t *testing.T) {
+		assert.NoError(t, clusters.CleanupGeneratedResources(ctx, env.Cluster(), t.Name()+"other"))
+	}(t)
+
+	t.Log("deploying a supported gatewayclass to the test cluster")
+	c, err := gatewayclient.NewForConfig(env.Cluster().Config())
+	require.NoError(t, err)
+	gwc := &gatewayv1alpha2.GatewayClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: uuid.NewString(),
+		},
+		Spec: gatewayv1alpha2.GatewayClassSpec{
+			ControllerName: gateway.ControllerName,
+		},
+	}
+	gwc, err = c.GatewayV1alpha2().GatewayClasses().Create(ctx, gwc, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	defer func() {
+		t.Log("cleaning up gatewayclasses")
+		if err := c.GatewayV1alpha2().GatewayClasses().Delete(ctx, gwc.Name, metav1.DeleteOptions{}); err != nil {
+			if !errors.IsNotFound(err) {
+				assert.NoError(t, err)
+			}
+		}
+	}()
+
+	t.Log("deploying a gateway that only allows routes in all namespaces")
```

Comment:
use of the word "only" here kinda seems off maybe?

```suggestion
	t.Log("deploying a gateway that allows routes in all namespaces")
```